src/Microsoft.TemplateEngine.Cli/New3Command.cs

```diff
@@ -567,10 +570,25 @@ private bool SyncOptionalWorkloads()
                         Installer.InstallPackages(owInstallationRequests);
                     }
 
-                    /* TODO: add comparison of Optional SDK Workload templates from Hive and returned by TemplateLocator
-                     * implement templates removal if they are not returned by API anymore
-                     */
+                    // remove uninstalled Optional SDK Workload packages
+                    List<string> owRemovalRequestsPackageIds = new List<string>();
+                    foreach(IInstallUnitDescriptor descriptor in _settingsLoader.InstallUnitDescriptorCache.Descriptors.Values)
```

Comment:
We already iterating through the descriptors, you can just add your code there (several lines above)

Reply:
No, I can't if I want to use a HashSet for sync requests' ids. But I can use <a>owInstalledPkgs.Keys</a> for iterating. Fixed this.